themes/_core/js/checkout-payment.js

```diff
@@ -146,6 +146,16 @@ class Payment {
       return;
     }
 
+    $.post(window.location.href, {
+      ajax: 1,
+      action: 'checkProductsStillOrderable',
+    })
+      .then((resp) => {
+        if (resp.allGood === false && resp.cartUrl !== '') {
+          location.href = resp.cartUrl;
```

Comment:
```suggestion
          location.href = resp.cartUrl;
```

I'm not sure about this, I think we should have an event here, and by default, we could redirect, but the developers should have a possibility to override it.

I would go with the
`prestashop.emit('orderConfirmationErrors', function () { // as much usable info as possible, selected carrier? payment method? }`

cc @NeOMakinG @Oksydan what do you think guys?

Reply:
Looks better yes, so it's overridable!

Reply:
@NeOMakinG Can you implement it into the PR for me? ;-)

Reply:
Ok, hope to have some time asap